packages/core/src/sagas/sendFilesToPostActivitySaga.js

```diff
@@ -9,16 +9,17 @@ import whileConnected from './effects/whileConnected';
 
 import { SEND_FILES } from '../actions/sendFiles';
 import postActivity from '../actions/postActivity';
-import stopSpeakingActivity from '../actions/stopSpeakingActivity';
 
 const getType = mime.getType.bind(mime);
 
 export default function* () {
   yield whileConnected(function* () {
-    for (;;) {
-      const { payload: { files } } = yield take(SEND_FILES);
-
-      if (files.length) {
+    yield takeEvery(
+      ({ payload, type }) => (
```

Comment:
inlining this takes away from the readability for me. can it be defined outside the function and given a good name?